### PR TITLE
Updated pettingzoo env to acomidate api changes and fixes

### PR DESCRIPTION
**Pull request by benblack769**
## Why are these changes needed?

Makes pettingzoo wrapper work well with upstream API changes in PettingZoo. Also allows wrapper to learn turn based games like Hanabi more easily. 

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by justinkterry** [comment]
This specifically is for supporting the changes added in PettingZoo 1.4.0

**Comment by justinkterry** [comment]
@ericl can you please merge this? the failing tests are unrelated to us

**Review by ericl — APPROVED** [review]

File: python/requirements.txt [hunk 1]
```diff
@@ -56,7 +56,7 @@ mypy
 networkx
 numba
 openpyxl
-pettingzoo>=1.3.2
+pettingzoo>=1.4.0
```
Comment (ericl) [inline python/requirements.txt:59]: To avoid always breaking travis when pettingzoo updates, should we pin this version in the future?

Comment (weepingwillowben) [reply python/requirements.txt:59]: Having a pinned version is fine as long as it is purely internal (which I believe it is here). Feel free to change this line to ==

**Pull request merged**